### PR TITLE
 Add node sleep task

### PR DESCRIPTION
**Pull request by pengz1**
* Node sleep task will set node to sleep for a duration

### CODE REVIEW

**Comment by pengz1** [comment]
@iceiilin @sunnyqianzhang @lanchongyizu

File: lib/task-data/tasks/node-sleep.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+'use strict';
+
+module.exports = {
+    friendlyName: 'Node sleep',
+    injectableName: 'Task.Node.Sleep',
+    implementsTask: 'Task.Base.Linux.Commands',
+    options: {
+        duration: 10, //in seconds
+        commands: ['sudo sleep {{options.duration}}']
```
Comment (yyscamper) [inline lib/task-data/tasks/node-sleep.js]: Why use the micro kernel sleep? isn't same to delay in RackHD. The bluebird has function `Promise.delay` to do delay in async mode.

File: lib/task-data/tasks/node-sleep.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+module.exports = {
+    friendlyName: 'Node sleep',
+    injectableName: 'Task.Node.Sleep',
+    implementsTask: 'Task.Base.Linux.Commands',
+    options: {
+        duration: 10, //in seconds
```
Comment (yyscamper) [inline lib/task-data/tasks/node-sleep.js]: Usually we prefer the delay in milliseconds.

File: lib/task-data/tasks/node-sleep.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2015, EMC, Inc.
```
Comment (yyscamper) [inline lib/task-data/tasks/node-sleep.js]: Copyright date is not correct.

File: spec/lib/task-data/tasks/node-sleep-spec.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2015, EMC, Inc.
```
Comment (yyscamper) [inline spec/lib/task-data/tasks/node-sleep-spec.js]: Incorrect copyright date

File: lib/task-data/tasks/node-sleep.js (hunk 1 above)
Comment (pengz1) [inline lib/task-data/tasks/node-sleep.js]: There is difference between RackHD delay and node delay. In the case a delay after reboot, RackHD delay starts from reboot finished while a node delay can only start after microkernel is ready.

**Comment by lanchongyizu** [comment]
:+1:

File: lib/task-data/tasks/node-sleep.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+'use strict';
+
+module.exports = {
+    friendlyName: 'Node sleep',
+    injectableName: 'Task.Node.Sleep',
+    implementsTask: 'Task.Base.Linux.Commands',
+    options: {
+        duration: 10000, //in milliseconds
+        commands: ['sudo sleep $(({{options.duration}}/1000))']
```
Comment (yyscamper) [inline lib/task-data/tasks/node-sleep.js]: This doesn't support float number calculation, for float number, you could try:
```
>>> $(echo "scale=3; 500/110" | bc)
4.545
>>> $(awk 'BEGIN{print 500/110 }')
4.54545
```

**Comment by sunnyqianzhang** [comment]
:+1:

File: lib/task-data/tasks/node-sleep.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+'use strict';
+
+module.exports = {
+    friendlyName: 'Node sleep',
+    injectableName: 'Task.Node.Sleep',
+    implementsTask: 'Task.Base.Linux.Commands',
+    options: {
+        duration: 10000, //in milliseconds
+        commands: ['sudo sleep $(awk "BEGIN{print {{options.duration}}/1000}")']
```
Comment (ytjohn) [inline lib/task-data/tasks/node-sleep.js]: Not a big deal, but we don't need privilege escalation for sleep.

Comment (pengz1) [inline lib/task-data/tasks/node-sleep.js]: @ytjohn 
This delay is configurable in secure, if you don't need delay, you can configure it to 0. You test reminds me we only have RackHD delay but don't have a real node delay, so I created this task that can also be used for other workflows.
Also after fixing secure erase job bug in:  https://github.com/RackHD/on-tasks/pull/416, You should be able achieve delay with noop task as you wanted.

**Pull request merged**